lib/reverse_sentence.rb

```diff
@@ -1,4 +1,52 @@
-# A method to reverse the words in a sentence, in place.
+# A method to reverse each word in a sentence, in place.
+def reverse_words(my_words)
```

Comment:
This method uses four temporary storage spaces to hold position, string_length, start_point and end_point. This is always true regardless of the input my_words (the value of it or the number of characters in it). Hence, the space complexity of this method is O(1) or constant.

To count the number of operations, we look at the nested loops. By going through one of the two inner, until loops, all n characters in my_words are examined and compared. Through the examination the beginning and end indices of a word is identified. Then, reverse_word is called - which reverses the characters in the word by examining the characters in the word one more time. Imagine a scenario where there is only one word in the input string. In this case, all n characters will be examined once to determine that the word begins at index 0 and ends at index string_length-1. Then, reverse_word reverses the characters in the word by swapping n/2 times. In this case there are n+(n/2) operations. Let's consider the other extreme, where there are only spaces in the string. In this case, all n characters will be examined once. reverse_word will not conduct any swaps since start_point will be greater than end_point. So, regardless, the number of operations will be O(n) or linear.